backend/msgpack/writer.go

```diff
@@ -209,6 +215,32 @@ func (w *writer) encodeWithLock(
 				PoliciesList: pl,
 			}
 			err = encoder.EncodeBatchTimerWithPoliciesList(btp)
+			if err != nil {
+				break
+			}
+
+			sizeAfter := buffer.Len()
+			if sizeAfter > w.flushSize {
```

Comment:
Perhaps reverse the condition and do `if sizeAfter < w.flushSize { continue } ...` to reduce nesting?